src/main/java/com/hubspot/imap/utils/parsers/fetch/EnvelopeParser.java

```diff
@@ -79,8 +112,49 @@ public Envelope parse(List<Object> in) {
     return envelope.build();
   }
 
+  public static Envelope fromHeader(Header header) {
```

Comment:
NIT: This should probably be named `parseHeader` cuz it exists in the `EnvelopParser` class (`fromHeader` should exist in the `Envelop` class)